src/probe/probe.c

```diff
@@ -961,7 +962,72 @@ static void probe_cb_free(void *arg, enum notify_id type, void *data)
 
 static bool probe_purpose_needs_ext_dma(uint32_t purpose)
 {
+#if CONFIG_IPC_MAJOR_4
+	return purpose == PROBE_PURPOSE_EXTRACTION;
+#else
 	return purpose == PROBE_PURPOSE_EXTRACTION || purpose == PROBE_PURPOSE_LOGGING;
+#endif
+}
+
+#if CONFIG_IPC_MAJOR_4
+static struct comp_buffer *ipc4_get_buffer(struct ipc_comp_dev *dev, probe_point_id_t probe_point)
+{
+	struct comp_buffer *buf;
+	struct list_item *sink_list, *source_list;
+
+	if (probe_point.fields.type == PROBE_TYPE_INPUT) {
+		if (list_is_empty(&dev->cd->bsource_list))
+			return NULL;
+
+		list_for_item(source_list, &dev->cd->bsource_list) {
+			buf = container_of(source_list, struct comp_buffer, sink_list);
+
```

Comment:
@lyakh   change ?
````c
struct comp_buffer __sparse_cache *buf_c;
buf_c = buffer_acquire(buf);
IPC4_SRC_QUEUE_ID(buf_c->id) ....
````

Reply:
yes, good catch! On both `if` branches of course